src/js/helpers/tasHelper.js

```diff
@@ -37,39 +37,18 @@ export const cleanTasData = (nodes) => cleanTreeData(nodes, tasKeyMap)
         return node;
     });
 
-export const getTasNodeFromTree = (tree, id) => {
-    let selectedNode;
-    tree
-        .forEach((agency) => {
-            if (agency.value === id) selectedNode = agency;
-        });
-    if (selectedNode) return selectedNode;
-    tree
-        .forEach((agency) => {
-            if (agency.children) {
-                agency.children
-                    .forEach((federalAccount) => {
-                        if (federalAccount.value === id) selectedNode = federalAccount;
-                    });
-            }
-        });
-    if (selectedNode) return selectedNode;
-    tree
-        .forEach((agency) => {
-            if (agency.children) {
-                agency.children
-                    .forEach((federalAccount) => {
-                        if (federalAccount.children) {
-                            federalAccount.children
-                                .forEach((tas) => {
-                                    if (tas.value === id) selectedNode = tas;
-                                });
-                        }
-                    });
-            }
-        });
-    return selectedNode;
+export const getTasNodeFromTree = (tree, code) => {
+    if (!tree) return tree;
```

Comment:
to me, this is confusing, and reason we return tree? i guess your returning null? i would rather see return null